### DISCUSSION TITLE
Complete all components of a path

### DISCUSSION THREAD

**Discussion by op06072** [marlonrichert/zsh-autocomplete#676] [Q&A]
## What do you want?
Hi I want directory autocomplete like this (/ho/op -> /home/op06072).

## Why do you want this?
I use this feature very often in terminal. Cause I can complete directory input with just one 'tap' click.

## Who else would benefit from this?
Everyone inputting directory

## How should it work?

Given the following situation:
* Everytime when I know full directory and input the directory

When I perform the following steps:
1. Input the incomplete directory (/ho/op)
2. Input 'tab' key
3. The directory is completed (/home/op06072)

Then I expect the following to happen:
* pressing down 'tab' after input directory make directory completed

Thanks for making awsome zsh plugin

**Comment by baudneo** [comment]
I'm pretty sure this is already implemented, I just tested it myself.

```
❯ ls -l /home/baudneo/
total 4128
-rwxr-xr-x 1 baudneo baudneo     743 Jul 23 12:18 authorized_keys
-rw------- 1 baudneo baudneo    3199 May  6  2023 baudneo.asc
drwxr-xr-x 1 baudneo baudneo      48 Oct 26 10:24 CLionProjects
drwxr-xr-x 1 baudneo baudneo       8 Oct 26 15:00 datumaro
drwxr-xr-x 1 baudneo baudneo    1038 Dec 11 12:50 Documents
drwx------ 1 baudneo baudneo    3054 Dec 12 22:00 Downloads
drwxr-xr-x 1 baudneo baudneo      94 Jun 27 22:29 emporia
-rw-r--r-- 1 baudneo baudneo    1078 Nov 13 09:08 file.py
-rw-r--r-- 1 baudneo baudneo 4194304 Jun 27 22:21 fw-backup-4M-emporia.bin
drwxr-xr-x 1 baudneo baudneo     272 Dec 13 21:46 git
-rwxr-xr-x 1 baudneo baudneo     307 Jul 23 13:34 healthcheck.io.sh
drwxr-xr-x 1 baudneo baudneo       0 Nov 11 17:27 intel
drwxr-xr-x 1 baudneo baudneo      38 Dec  9 10:53 Library
-rw-r--r-- 1 baudneo baudneo      76 Dec 10 17:38 minicom.log
drwxr-xr-x 1 baudneo baudneo    1428 Dec  3 23:25 Pictures
drwxr-xr-x 1 baudneo baudneo      10 Jul 30 11:26 Postman
drwxr-xr-x 1 baudneo baudneo       6 Nov 13 12:46 projects
-rw-r--r-- 1 baudneo baudneo     435 Nov 17 18:16 pub_key
drwxr-xr-x 1 baudneo baudneo     546 Dec  9 12:02 PycharmProjects
drwxr-xr-x 1 baudneo baudneo     370 Jul 23 22:38 sg_logs
-rw-r--r-- 1 baudneo baudneo     402 Dec 11 15:35 test.fio
drwxr-xr-x 1 baudneo baudneo      90 Nov 13 14:16 ultralytics-dfo-dataset
drwxr-xr-x 1 baudneo baudneo     120 Dec 12 22:00 Videos
drwxr-xr-x 1 baudneo baudneo      14 Jun  6  2023 WebstormProjects
-rwxr-xr-x 1 baudneo baudneo    2494 Jul 22 23:08 wz_mini.conf
drwxr-xr-x 1 baudneo baudneo     168 Jul 22 23:10 wz_mini_hacks
drwxr-xr-x 1 baudneo baudneo     246 Oct  9 11:30 zm

    ~/git/fss    main ···················································································· 2.00   26.3G  ─╮
❯ cd /home/baudneo/emp
```
When I hit \<TAB\>, it auto completed emporia for me.

If there are multiple choices, it lists them for me and allows me to use the arrow keys to select which completion I want (or I can keep typing and hit \<TAB\> again to get a completion). Below is an example of it giving me a zsh menu I can navigate with arrow keys to select which dir. `cd /home/baudneo/<TAB>`

```
❯ cd /home/baudneo/                                        ⇣40.1 KiB/s ⇡2.94 KiB/s 10.0.1.151   09:01:11 PM  ─╯
Completing directory
CLionProjects/            emporia/                  Pictures/                 sg_logs/                  wz_mini_hacks/
datumaro/                 git/                      Postman/                  ultralytics-dfo-dataset/  zm/
Documents/                intel/                    projects/                 Videos/
Downloads/                Library/                  PycharmProjects/          WebstormProjects/
```

**Comment by op06072** [comment]
> I'm pretty sure this is already implemented, I just tested it myself.
> 
> ```
> ❯ ls -l /home/baudneo/
> total 4128
> -rwxr-xr-x 1 baudneo baudneo     743 Jul 23 12:18 authorized_keys
> -rw------- 1 baudneo baudneo    3199 May  6  2023 baudneo.asc
> drwxr-xr-x 1 baudneo baudneo      48 Oct 26 10:24 CLionProjects
> drwxr-xr-x 1 baudneo baudneo       8 Oct 26 15:00 datumaro
> drwxr-xr-x 1 baudneo baudneo    1038 Dec 11 12:50 Documents
> drwx------ 1 baudneo baudneo    3054 Dec 12 22:00 Downloads
> drwxr-xr-x 1 baudneo baudneo      94 Jun 27 22:29 emporia
> -rw-r--r-- 1 baudneo baudneo    1078 Nov 13 09:08 file.py
> -rw-r--r-- 1 baudneo baudneo 4194304 Jun 27 22:21 fw-backup-4M-emporia.bin
> drwxr-xr-x 1 baudneo baudneo     272 Dec 13 21:46 git
> -rwxr-xr-x 1 baudneo baudneo     307 Jul 23 13:34 healthcheck.io.sh
> drwxr-xr-x 1 baudneo baudneo       0 Nov 11 17:27 intel
> drwxr-xr-x 1 baudneo baudneo      38 Dec  9 10:53 Library
> -rw-r--r-- 1 baudneo baudneo      76 Dec 10 17:38 minicom.log
> drwxr-xr-x 1 baudneo baudneo    1428 Dec  3 23:25 Pictures
> drwxr-xr-x 1 baudneo baudneo      10 Jul 30 11:26 Postman
> drwxr-xr-x 1 baudneo baudneo       6 Nov 13 12:46 projects
> -rw-r--r-- 1 baudneo baudneo     435 Nov 17 18:16 pub_key
> drwxr-xr-x 1 baudneo baudneo     546 Dec  9 12:02 PycharmProjects
> drwxr-xr-x 1 baudneo baudneo     370 Jul 23 22:38 sg_logs
> -rw-r--r-- 1 baudneo baudneo     402 Dec 11 15:35 test.fio
> drwxr-xr-x 1 baudneo baudneo      90 Nov 13 14:16 ultralytics-dfo-dataset
> drwxr-xr-x 1 baudneo baudneo     120 Dec 12 22:00 Videos
> drwxr-xr-x 1 baudneo baudneo      14 Jun  6  2023 WebstormProjects
> -rwxr-xr-x 1 baudneo baudneo    2494 Jul 22 23:08 wz_mini.conf
> drwxr-xr-x 1 baudneo baudneo     168 Jul 22 23:10 wz_mini_hacks
> drwxr-xr-x 1 baudneo baudneo     246 Oct  9 11:30 zm
> 
>     ~/git/fss    main ···················································································· 2.00   26.3G  ─╮
> ❯ cd /home/baudneo/emp
> ```
> When I hit \<TAB\>, it auto completed emporia for me.
> 
> If there are multiple choices, it lists them for me and allows me to use the arrow keys to select which completion I want (or I can keep typing and hit \<TAB\> again to get a completion). Below is an example of it giving me a zsh menu I can navigate with arrow keys to select which dir. `cd /home/baudneo/<TAB>`
> 
> ```
> ❯ cd /home/baudneo/                                        ⇣40.1 KiB/s ⇡2.94 KiB/s 10.0.1.151   09:01:11 PM  ─╯
> Completing directory
> CLionProjects/            emporia/                  Pictures/                 sg_logs/                  wz_mini_hacks/
> datumaro/                 git/                      Postman/                  ultralytics-dfo-dataset/  zm/
> Documents/                intel/                    projects/                 Videos/
> Downloads/                Library/                  PycharmProjects/          WebstormProjects/
> ```

I mean not just autocomplete last part.

**Comment by marlonrichert** [comment]
Autocomplete disables this by default, because it used to be that, if you tried to complete a non-existing path this way, it could make completion hang for a very long time. However, I just tested it with the latest version of Autocomplete and this no longer appears to be the case. I will add it back: #677
